src/m365/teams/commands/cache/cache-remove.spec.ts

```diff
@@ -0,0 +1,267 @@
+
+    promptOptions = undefined;
+
+    sinon.stub(Cli, 'prompt').callsFake((options: any, cb: (result: { continue: boolean }) => void) => {
+      promptOptions = options;
+      cb({ continue: true });
+    });
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      Cli.prompt
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      appInsights.trackEvent,
+      Cli.getInstance().config.all
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name.startsWith(commands.CACHE_REMOVE), true);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('prompts before clear cache when confirm option not passed', (done) => {
+    sinon.stub(process, 'platform').value('win32');
+    sinon.stub(process, 'env').value({ 'CLIMICROSOFT365_ENV': '' });
+
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake((options: any, cb: (result: { continue: boolean }) => void) => {
+      promptOptions = options;
+      cb({ continue: false });
+    });
+
+    command.action(logger, {
+      options: { }
+    }, () => {
+      let promptIssued = false;
+
+      if (promptOptions && promptOptions.type === 'confirm') {
+        promptIssued = true;
+      }
+
+      try {
+        assert(promptIssued);
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+  });
+
+  it('fails validation if called from docker container.', (done) => { 
+    sinon.stub(process, 'platform').value('win32');
+    sinon.stub(process, 'env').value({ 'CLIMICROSOFT365_ENV': 'docker' });
+    
+    const actual = command.validate({
+      options: { }
+    });
+    assert.notStrictEqual(actual, true);
+    done();
+  });
+
+  it('fails validation if not called from win32 or darwin platform.', (done) => {
+    sinon.stub(process, 'platform').value('android');
+    sinon.stub(process, 'env').value({ 'CLIMICROSOFT365_ENV': '' });
+    
+    const actual = command.validate({
+      options: { }
+    });
+    assert.notStrictEqual(actual, true);
+    done();
+  });
+
+  it('passes validation if called from win32 or darwin platform.', (done) => {
+    sinon.stub(process, 'platform').value('win32');
+    sinon.stub(process, 'env').value({ 'CLIMICROSOFT365_ENV': '' });
+    
+    const actual = command.validate({
+      options: { }
+    });
+    assert.strictEqual(actual, true);
+    done();
+  });
+
+  it('fails to remove teams cache when exec fails', (done) => {
+    sinon.stub(process, 'platform').value('win32');
+    sinon.stub(process, 'env').value({ 'CLIMICROSOFT365_ENV': '' });
+    const error = new Error('ERROR: The process "Teams.exe" not found.');
+    const exec = sinon.stub(command, 'exec' as any).throws(error);
+
+    command.action(logger, { options: {
+      confirm: true,
+      verbose: true
+    }}, () => {
+      try {
+        assert(true);
+        exec.restore();
```

Comment:
We should do this in the `finally` clause of `try..catch` in case the assert fails